pkg/proxy/conntrack/conntrack_test.go

```diff
@@ -77,204 +93,177 @@ func TestExecConntrackTool(t *testing.T) {
-			func(cmd string, args ...string) exec.Cmd { return fakeexec.InitFakeCmd(&fcmd, cmd, args...) },
-			func(cmd string, args ...string) exec.Cmd { return fakeexec.InitFakeCmd(&fcmd, cmd, args...) },
-		},
-		LookPathFunc: func(cmd string) (string, error) { return cmd, nil },
-	}
-
+func TestClearEntriesForIP(t *testing.T) {
 	testCases := []struct {
 		name string
 		ip   string
+
+		expectCommand string
 	}{
-		{"IPv4 success", "10.240.0.3"},
-		{"IPv4 success", "10.240.0.5"},
-		{"IPv4 simulated error", "10.240.0.4"},
-		{"IPv6 success", "2001:db8::10"},
+		{
+			name: "IPv4",
+			ip:   "10.240.0.3",
+
+			expectCommand: "conntrack -D --orig-dst 10.240.0.3 -p udp",
+		},
+		{
+			name: "IPv6",
+			ip:   "2001:db8::10",
+
+			expectCommand: "conntrack -D --orig-dst 2001:db8::10 -p udp -f ipv6",
+		},
 	}
 
-	svcCount := 0
 	for _, tc := range testCases {
+		fexec, fcmd := makeCT(success)
 		if err := ClearEntriesForIP(fexec, tc.ip, v1.ProtocolUDP); err != nil {
-			t.Errorf("%s test case:, Unexpected error: %v", tc.name, err)
+			t.Errorf("%s/success:, Unexpected error: %v", tc.name, err)
 		}
-		expectCommand := fmt.Sprintf("conntrack -D --orig-dst %s -p udp", tc.ip) + familyParamStr(utilnet.IsIPv6String(tc.ip))
-		execCommand := strings.Join(fcmd.CombinedOutputLog[svcCount], " ")
-		if expectCommand != execCommand {
-			t.Errorf("%s test case: Expect command: %s, but executed %s", tc.name, expectCommand, execCommand)
+		execCommand := getExecutedCommand(fexec, fcmd)
+		if tc.expectCommand != execCommand {
+			t.Errorf("%s/success: Expect command: %s, but executed %s", tc.name, tc.expectCommand, execCommand)
 		}
-		svcCount++
-	}
-	if svcCount != fexec.CommandCalls {
-		t.Errorf("Expect command executed %d times, but got %d", svcCount, fexec.CommandCalls)
-	}
-}
 
-func TestClearUDPConntrackForPort(t *testing.T) {
-	fcmd := fakeexec.FakeCmd{
-		CombinedOutputScript: []fakeexec.FakeAction{
-			func() ([]byte, []byte, error) { return []byte("1 flow entries have been deleted"), nil, nil },
-			func() ([]byte, []byte, error) {
-				return []byte(""), nil, fmt.Errorf("conntrack v1.4.2 (conntrack-tools): 0 flow entries have been deleted")
-			},
-			func() ([]byte, []byte, error) { return []byte("1 flow entries have been deleted"), nil, nil },
-		},
-	}
-	fexec := &fakeexec.FakeExec{
-		CommandScript: []fakeexec.FakeCommandAction{
-			func(cmd string, args ...string) exec.Cmd { return fakeexec.InitFakeCmd(&fcmd, cmd, args...) },
-			func(cmd string, args ...string) exec.Cmd { return fakeexec.InitFakeCmd(&fcmd, cmd, args...) },
-			func(cmd string, args ...string) exec.Cmd { return fakeexec.InitFakeCmd(&fcmd, cmd, args...) },
-		},
-		LookPathFunc: func(cmd string) (string, error) { return cmd, nil },
+		fexec, _ = makeCT(nothingToDelete)
+		if err := ClearEntriesForIP(fexec, tc.ip, v1.ProtocolUDP); err != nil {
+			t.Errorf("%s/nothing to delete:, Unexpected error: %v", tc.name, err)
+		}
```

Comment:
ok, this simulates that the second execution of the same parameters should not find anything to delete